src/cdi_CPEloss/Analytic_Spitzer_Eloss_Model.cc

```diff
@@ -0,0 +1,69 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_CPEloss/Analytic_Spitzer_Eloss_Model.cc
+ * \author Ben R. Ryan
+ * \date   Feb 21 2020
+ * \brief  Analytic_Spitzer_Eloss_Model implementation file.
+ * \note   Copyright (C) 2016-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "Analytic_Spitzer_Eloss_Model.hh"
+#include "ds++/Assert.hh"
+
+namespace rtt_cdi_cpeloss {
+
+/*! \brief Calculate the eloss in units of keV shk^-1; T given in keV,
+ *         rho in g/cc, v0 in cm/shk
+ *
+ * CP energy loss class using standard free particle approach to stopping power
+ * with Spitzer prescription for Coulomb logarithm.
+ *
+ * \param T material temperature in keV
+ * \param rho material density in g/cm^3
+ * \param v incident particle speed in cm/shk
+ * \return eloss (vector of time coefficients) in keV shk^-1
+ */
+double Analytic_Spitzer_Eloss_Model::calculate_eloss(const double T,
+                                                     const double rho,
+                                                     const double v) const {
+  Require(T >= 0.0);
+  Require(rho >= 0.0);
+  Require(v >= 0.0);
+
+  const double keV = 1e3 * pc.eV();
+  const double eloss_unit_fac = 1.e-8 / keV;
+
+  const double nt = rho / mt;
+  const double Tt = T * keV / pc.k();
+  const double vp = v * 1.e8;
+  const double vt = sqrt(2. * pc.k() * Tt / mt);
+  const double Ep = 1. / 2. * mp * vp * vp;
+  const double eloss_max = 1e3 * keV * nt;
```

Comment:
Is this cap specified by Spitzer, or Ben Ryan? Just curious :)

Reply:
I just copied this from what's used for a practically-infinite stopping power in some tabulated data -- since the units don't make sense I'll define a new `rtt_cdi::constants::max_eloss` up in the `CPCommon` to be used throughout.